### PR TITLE
4219: implement the "power" scalar function

### PR DESCRIPTION
**Pull request by andreidan**
Power function implementation and docs

### CODE REVIEW

File: sql/src/test/java/io/crate/operation/scalar/arithmetic/PowerFunctionTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+package io.crate.operation.scalar.arithmetic;
+
+import io.crate.operation.scalar.AbstractScalarFunctionsTest;
+import org.junit.Test;
+
+public class PowerFunctionTest extends AbstractScalarFunctionsTest {
+
+    @Test
+    public void testPowerWithIntegers() {
+        assertEvaluate("power(2,4)", 16.0);
```
Comment (dobe) [inline sql/src/test/java/io/crate/operation/scalar/arithmetic/PowerFunctionTest.java]: this should evaluate to an integral value (16)

as an example see https://github.com/crate/crate/blob/master/sql/src/main/java/io/crate/operation/scalar/arithmetic/MultiplyFunction.java

Comment (andreidan) [inline sql/src/test/java/io/crate/operation/scalar/arithmetic/PowerFunctionTest.java]: I think the multiply operation is slightly different, in that if the both the input values are of an integral type, the result will be of an integral type.

The same doesn't hold for the power function as a negative exponent will yield a decimal type result. If we want to maintain the type of the inputs (integrals) then we will have to round the result - ie: 2^-3=0, 2^-1=1. 
Or in this case, if the result contains a decimal type, shall we change the type of the result to double (this feels non intuitive to me), but keep it integral for the other cases?

File: blackbox/docs/sql/scalar.txt [hunk 2]
```diff
@@ -749,6 +749,19 @@ The second argument (``b``) is optional. If not present, base 10 is used::
...
     e.g. log(10, 1) does.
 
+power(a: number, b: number) returns double
+--------------------------------------------------
+Returns the given argument ``a`` raised to the power of argument ``b``.
+
+See below for an example:
+
+    cr> select power(2,3) from sys.cluster;
+    +---------------+
```
Comment (matriv) [inline blackbox/docs/sql/scalar.txt]: Please fix output table. See: https://travis-ci.org/crate/crate/jobs/166565128#L1977
You can run these tests with: ./gradlew itest

**Comment by mfussenegger** [comment]
Cool stuff. Merged it to master (https://github.com/crate/crate/commit/1ab91f127c8bb1ec1effa05e183975935a6206cc)

**Pull request closed without merging**